name.abuchen.portfolio/src/name/abuchen/portfolio/datatransfer/pdf/ComdirectPDFExtractor.java

```diff
@@ -1155,170 +1728,332 @@ else if (v.get("note1").matches("^(?i:Visa\\-Kartenabre.*)$"))
+
+        // Filter transactions by dividend transactions
+        List<Item> dividendTransactionList = items.stream() //
+                        .filter(TransactionItem.class::isInstance) //
+                        .map(TransactionItem.class::cast) //
+                        .filter(i -> i.getSubject() instanceof AccountTransaction) //
+                        .filter(i -> AccountTransaction.Type.DIVIDENDS //
+                                        .equals((((AccountTransaction) i.getSubject()).getType()))) //
+                        .collect(Collectors.toList());
+
+        // Group sell and taxes transactions together and group by date and security
+        Map<LocalDateTime, Map<Security, List<Item>>> sellTaxesTransactions;
+        if (!sellTransactionList.isEmpty())
+            sellTaxesTransactions = Stream.concat(sellTransactionList.stream(), taxesTransactionList.stream())
+                            .collect(Collectors.groupingBy(Item::getDate, Collectors.groupingBy(Item::getSecurity)));
+        else
+            sellTaxesTransactions = Collections.emptyMap();
 
-            if (isDividendTransaction(i))
-            {
-                List<Item> similarTransactions = dividends.get(i.getDate()).get(i.getSecurity());
+        // Group dividend and taxes transactions together and group by date and security
+        Map<LocalDateTime, Map<Security, List<Item>>> dividendTaxesTransactions;
+        if (!dividendTransactionList.isEmpty())
+            dividendTaxesTransactions = Stream.concat(dividendTransactionList.stream(), taxesTransactionList.stream())
+                            .collect(Collectors.groupingBy(Item::getDate, Collectors.groupingBy(Item::getSecurity)));
+        else
+            dividendTaxesTransactions = Collections.emptyMap();
 
-                // Are there multiple dividend transactions?
-                if (similarTransactions.size() == 2)
-                {
-                    AccountTransaction a1 = (AccountTransaction) similarTransactions.get(0).getSubject();
-                    AccountTransaction a2;
-                    int ownIndex = 0;
+        sellTaxesTransactions.forEach((k, v) -> {
+            v.forEach((security, transactions) -> {
 
-                    // a1 = self, a2 = other
-                    if (i.equals(similarTransactions.get(0)))
+                if (transactions.size() == 1)
+                {
+                    if (transactions.get(0).getSubject() instanceof BuySellEntry portfolioTransaction)
                     {
-                        a2 = (AccountTransaction) similarTransactions.get(1).getSubject();
+                        BuySellEntry sellTransaction = (BuySellEntry) transactions.get(0).getSubject();
+
+                        // Set the time to the transaction date if available
+                        if (sellTransaction.getPortfolioTransaction().getSecurity().getAttributes()
+                                        .get(new AttributeType(ATTRIBUTE_DATE_TIME)) != null)
```

Comment:
I understand the reasoning: the sale document contains a time, the tax document does not.

However, storing the date time as attribute does not work for multiple reasons: first: what happens if there are multiple sale transactions for one security - which date/time is winning? Second: the attributes are polluting the XML.

Maybe a better way is to create the grouping not on LocalDateTime but on LocalDate. In the process, it is "just" stripping the time. But the transaction keeps the original time. Something along these lines:

```java
// Group dividend and taxes transactions together and group by date and security
Map<LocalDate, Map<Security, List<Item>>> dividendTaxesTransactions;
[...]
dividendTaxesTransactions = Stream.concat(dividendTransactionList.stream(), taxesTransactionList.stream())
   .collect(Collectors.groupingBy(item -> item.getDate().toLocalDate(), Collectors.groupingBy(Item::getSecurity)));
```